MiniApp/Classes/MiniAppAdmobDisplayer.swift

```diff
@@ -0,0 +1,89 @@
+import Foundation
+import GoogleMobileAds
+
+/// Made to be a singleton
+/// It has to be a single instance to avoid multiple instances of Admob SDK
+class MiniAppAdmobDisplayer: NSObject, MiniAppAdDisplayProtocol {
+	var interstitial: GADInterstitial!
+	var rewarded: GADRewardedAd?
+	var onInterstitialClosed: (() -> Void)?
+	var onRewardedClosed: ((MiniAppReward?) -> Void)?
+	var lastReward: MiniAppReward?
+
+	override init() {
+		super.init()
+
+		GADMobileAds.sharedInstance().start(completionHandler: { [weak self] _ in
```

Comment:
The initialization of the Google Ads SDK should be done by the Host App. In the SDK, we can't know whether the Host App has already initialized the Ads SDK or not.